src/client.rs

```diff
@@ -76,7 +76,15 @@ impl fmt::Debug for Client {
     }
 }
 
-#[cfg(feature = "default-client")]
+#[cfg(all(
+    feature = "default-client",
+    any(
```

Comment:
These should not be necessary - we allow building the wasm client even if the target arch is not strictly `wasm32`.

Also, even if it was, most of these clients explicitly set the `default-client` feature, and so the casing would only need to be for the wasm client.

Reply:
> we allow building the wasm client even if the target arch is not strictly `wasm32`.

[http-client's wasm module compiles only on wasm targets](https://github.com/http-rs/http-client/blob/faf1b4a0dd68b82e0f6546b30df0de07f6918ee1/src/lib.rs#L22), so building wasm-client on non-wasm targets will fail to compile.

> Also, even if it was, most of these clients explicitly set the `default-client` feature, and so the casing would only need to be for the wasm client.

The problem is when the `default-client` feature is specified and no client is specified.
In that case, `DefaultClient` is not defined, so the compilation will fail if only `cfg(feature = "default-client")` is used.

https://github.com/http-rs/surf/blob/fc75a59cf8486888ff3ea0c8a5cfa5cf89391bf6/src/client.rs#L10-L20

However, `cfg(feature = "default-client")` isn't really needed here, so it seems right to remove it.

Reply:
> The problem is when the `default-client` feature is specified and no client is specified.
> In that case, `DefaultClient` is not defined, so the compilation will fail if only `cfg(feature = "default-client")` is used.

Another way to deal with this is to explicitly fail the compilation if only the `default-client` feature is specified: https://github.com/rust-lang/futures-rs/blob/c7328778999f8d20f5c3bb821a8ad38b72923ff9/futures-core/src/lib.rs#L13-L14
It may be better.